lib/kauth/save.go

```diff
@@ -0,0 +1,45 @@
+package kauth
+
+import (
+	"fmt"
+	"github.com/enfabrica/enkit/lib/cache"
+	"github.com/enfabrica/enkit/lib/kcerts"
+	"github.com/enfabrica/enkit/lib/logger"
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/ssh"
+	"time"
+)
+
+var NoSSHCredentialsErr = errors.New("no credentials passed in")
+// SaveCredentials saves the passed in credentials to the current ssh-agent. If the credentials are empty, i.e.
+// the EnkitCredentials only contain EnkitCredentials.Token, it will return NoSSHCredentialsErr.
+func SaveCredentials(credentials *EnkitCredentials, store cache.Store, l logger.Logger) error{
```

Comment:
thinking aloud... should this be a method on EnkitCredentials? Rather than a free standing function?

Reply:
either way is fine, do it or don't at your leisure.

Reply:
I would rather this be a function, just stylistically. Also means we don't need to go through the hoopla of embedding interfaces in the EnCreds